### ISSUE TITLE
Circular imports with import chain of more than 2 modules are not detected anymore

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3867]
### Discussed in https://github.com/microsoft/pyright/discussions/3865

<div type='discussions-op-text'>

<sup>Originally posted by **mailys-ds** August 26, 2022</sup>
Hello,

It seems that since 1.1.242, pyright doesn't detect circular import in chains that are made up of more than 2 modules.

To reproduce:
``` python
# module_a.py
from .module_b import B
class A():
    def foo(self, b: B) -> B:
        return b

# module_b.py
from .module_c import C
class B():
    def foo(self, c: C) -> C:
        return c

# module_c.py
from .module_a import A
class C():
    def foo(self, a: A) -> A:
        return a
```
This circular import used to be detected in pyright < 1.1.242. So is this behaviour expected or is it a bug ?

Thanks,
Maïlys.</div>

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.269, which I just published. It will also be included in a future version of pylance.
